### PR TITLE
[NU-1513] search panel

### PR DESCRIPTION
**Pull request by JulianWielga**
## Describe your changes

[NU-1513]

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

[NU-1513]: https://touk-jira.atlassian.net/browse/NU-1513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: designer/client/src/components/toolbars/search/utils.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    return useMemo(
+        () =>
+            allNodes
+                .map((node) => {
+                    const edges = allEdges
+                        .filter((e) => e.from === node.id)
+                        .filter((e) => matchFilters(e.edgeType?.condition, filterValues));
+
+                    const groups = findFields(filterValues, node)
+                        .concat(edges.length ? "edgeExpression" : null)
```
Comment (Dzuming) [inline designer/client/src/components/toolbars/search/utils.ts:100]: If i understand correctly, edgeExpression is also a field, but it's not a part of findFields logic. Maybe it's worth to have this findFields logic in one place, and move edgeExpression calculation inside of the findFIelds?

**Review by Dzuming — CHANGES_REQUESTED** [review]

File: designer/client/src/components/graph/node-modal/NodeDetailsContent.tsx [hunk 2]
```diff
@@ -44,6 +48,7 @@ export const NodeDetailsContent = ({
                 />
             </TestResultsWrapper>
             <NodeAdditionalInfoBox node={node} />
+            {userSettings["debug.nodesAsJson"] && <DebugNodeInspector node={node} />}
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply designer/client/src/components/graph/node-modal/NodeDetailsContent.tsx:51]: starting in dev mode shows new panel/toolbar. just add flag there.

File: designer/client/src/components/graph/Graph.tsx [hunk 3]
```diff
@@ -29,6 +29,12 @@ import { isTouchEvent, LONG_PRESS_TIME } from "../../helpers/detectDevice";
 import { batchGroupBy } from "../../reducers/graph/batchGroupBy";
 import { createUniqueArrowMarker } from "./arrowMarker";
 import { Scenario } from "../Process/types";
+import { nodeFocused, nodeValidationError } from "./focusableStyled";
+import { dragHovered } from "./GraphStyled";
+import { isEdgeConnected } from "./GraphPartialsInTS/EdgeUtils";
+
+// TODO: this is needed here due to our webpack config - needs fixing.
+styles;
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply designer/client/src/components/graph/Graph.tsx:37]: Added issue for that (NU-1559)

File: designer/client/src/components/toolbars/search/utils.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import NodeUtils from "../../graph/NodeUtils";
+import { useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { ensureArray } from "../../../common/arrayUtils";
+
+type SelectorResult = { expression: string } | string;
+type Selector = (node: NodeType) => SelectorResult | SelectorResult[];
+type FilterSelector = { name: string; selector: Selector }[];
+
+const fieldsSelectors: FilterSelector = [
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply designer/client/src/components/toolbars/search/utils.ts:14]: this is for next stages if ever

**Review by Dzuming — APPROVED** [review]

**Pull request merged**